projects/baby_driver/test/test_spi_exchange.c

```diff
@@ -57,6 +61,22 @@ static SpiSettings s_spi_settings_test_2 = { .baudrate = 6000000,
                                              .cs = CONTROLLER_BOARD_ADDR_SPI2_NSS };
 #define SPI_PORT_TEST_2 (SPI_PORT_2)
 
+static void prv_tx_rx_callback_msgs(uint16_t num_messages) {
+  // The ordering of events from babydriver callbacks is abnormal: CAN_TRANSMIT_BABYDRIVER
+  // is called in a rapidly firing callback, without event processing in between. So, we
+  // delay to let the callbacks fire, then process all TX events at once then all RX events
+  // at once.
+
+  // The +1 is to account for the initial delay before the module starts TXing
+  delay_us((uint32_t)(TEST_SPI_EXCHANGE_DELAY_US * (num_messages + 1)));
+  for (uint16_t i = 0; i < num_messages; i++) {
+    MS_TEST_HELPER_CAN_TX(TEST_CAN_EVENT_TX);
+  }
+  for (uint16_t i = 0; i < num_messages; i++) {
+    MS_TEST_HELPER_CAN_RX(TEST_CAN_EVENT_RX);
+  }
```

Comment:
This is still flaky on my machine - I think it might just be the very short length of the delay, causing the actual code execution to take a non trivial amount of time. On my machine lengthening `TEST_SPI_EXCHANGE_DELAY_US` by a couple orders of magnitude fixed it.

Reply:
Hmm, I changed it back to what it was originally. Since that's now a soft timer issue not a CAN issue I'll address it soon, but not here